### ISSUE TITLE
RunOutputCable fails if it has to copy a Dataset from the database to the sandbox

### ISSUE THREAD

**Issue by rhliang** [cfe-lab/Kive#558]
One of the new checks we've added to execution is that an input cable checks the integrity of a Dataset that it's responsible for copying into the sandbox.  Correspondingly there is now a field on `IntegrityCheckLog` called `runsic` for the RunSIC that invoked it, if applicable.  It turns out that a RunOutputCable can also be responsible for copying something into the sandbox; if you stop a Run after a step has finished but before one of the output cables coming from that step has run, and then you re-run that Run, when it reaches that output cable it will run it, and then execution fails because it then attempts to set the IntegrityCheckLog's `runsic` field to a RunOutputCable.

Two ideas for fixing this are:

* change the IntegrityCheckLog's `runsic` field to allow any RunComponent.  This way, whatever RunComponent is responsible for copying a Dataset into the sandbox can be assigned here.  At the moment I don't believe RunSteps can do this, so this would add a bit of redundancy.
* add a `runoutputcable` field to IntegrityCheckLog instead.

**Comment by donkirkby** [comment]
I like the first option, @rhliang. Calling the field `checked_by` would also make the relationship clearer.

**Comment by rhliang** [comment]
I like the first option too.  I might prefer to call the field `sandboxed_by` or `copied_into_place_by` (that's a bit wordy) or something similar to distinguish that this is the RunComponent that introduced it into a sandbox, rather than something that's checked it during the course of execution (i.e. it got regenerated).  Thoughts?

**Comment by rhliang** [comment]
Two other ideas: `loaded_by` or just `copied_by`

**Comment by donkirkby** [comment]
Your suggestions seem to be named from the dataset's perspective, but this is a field on `IntegrityCheckLog`, right? The integrity check was started by a run component, so how about calling the field `IntegrityCheckLog.started_by`?
Of course, we might be over thinking this. We could just call it `IntegrityCheckLog.pizza_jazz`.

![XKCD car model names, including the Acura PizzaJazz][xkcd]

[xkcd]: http://imgs.xkcd.com/comics/car_model_names.png

**Comment by rhliang** [comment]
I prefer `started_by` over `pizza_jazz`.
